stdlib/SuiteSparse/src/cholmod.jl

```diff
@@ -798,13 +798,25 @@ get_perm(FC::FactorComponent) = get_perm(Factor(FC))
 # Conversion/construction
 function Dense{T}(A::StridedVecOrMat) where T<:VTypes
     d = allocate_dense(size(A, 1), size(A, 2), stride(A, 2), T)
-    s = unsafe_load(pointer(d))
-    for i in eachindex(A)
-        unsafe_store!(s.x, A[i], i)
+    GC.@preserve d begin
+        s = unsafe_load(pointer(d))
+        for (i, c) in enumerate(eachindex(A))
+            unsafe_store!(s.x, A[c], i)
+        end
     end
     d
 end
-function Dense(A::StridedVecOrMat)
+function Dense{T}(A::Union{Adjoint{<:Any, <:StridedVecOrMat}, Transpose{<:Any, <:StridedVecOrMat}}) where T<:VTypes
+    d = allocate_dense(size(A, 1), size(A, 2), size(A, 1), T)
+    GC.@preserve d begin
+        s = unsafe_load(pointer(d))
+        for (i, c) in enumerate(eachindex(A))
+            unsafe_store!(s.x, A[c], i)
+        end
+    end
+    d
+end
```

Comment:
It seems to me that these two methods are exactly the same. So you could introduce
```julia
const StridedVecOrMatInclAdjAndTrans = Union{StridedVecOrMat, Adjoint{<:Any, <:StridedVecOrMat}, Transpose{<:Any, <:StridedVecOrMat}}
```
around here, and have the method as
```julia
function Dense{T}(A::StridedVecOrMatInclAdjAndTrans) where T<:VTypes
```